projects/plugins/protect/src/js/data/scan/use-fixers-mutation.ts

```diff
@@ -16,6 +16,20 @@ export default function useFixersMutation(): UseMutationResult {
 	return useMutation( {
 		mutationFn: API.fixThreats,
 		onSuccess: data => {
+			// Handle a top level error
+			if ( data?.error ) {
+				throw new Error( data.error );
+			}
+
+			const isThreatLevelError = Object.values( data?.threats ).every(
+				( threat: { error?: string } ) => Boolean( threat?.error )
+			);
+
+			// Handle a threat level error
+			if ( isThreatLevelError ) {
+				throw new Error();
+			}
```

Comment:
We're only throwing here when every threats has an error, and not passing back any of the error information. What do you think about throwing the first threat-level error that it finds in `data.threats`?

Reply:
We want to only throw an error if all threats we are checking have an error. We do not want to throw an error if the response contains valid fixer data, eg if we are checking two or more threats and one has an `in_progress` status the other has an error. If we throw an error any time we detect the error property, we will not set the fixer query state, and the UI will not show a fixer in progress even though there is and we will not poll fixer status.

Reply:
Last question - since we are handling the individual threat-level errors elsewhere, i.e. showing an error message via `showBulkNotices( successes, failures )`, why do we need to throw this generic error when all threats have errors?

Reply:
IIRC if we don't handle this possibility here with the initial data returned (from queuing the fixers), the UI wont update and we wont show an error unless we repoll fixer status and handle the return in the `useFixersQuery` hook, which we wont do because none of the data matches the scenario that will cause that.